packages/frontend/src/stdlib/analyses/schema_erd.tsx

```diff
@@ -0,0 +1,263 @@
+import { createResource, createSignal, Show } from "solid-js";
+
+import { Foldable, FormGroup, SelectField } from "catcolab-ui-components";
+import type { DblModel, MorType, ObType } from "catlog-wasm";
+import type { ModelAnalysisProps } from "../../analysis";
+import { DownloadSVGButton, loadViz } from "../../visualization";
+import { Direction, type SchemaERDConfig } from "./schema_erd_config";
+
+/** Visualize a schema as an Entity-Relationship Diagram.
+
+This visualization is specifically designed for schemas (models of the simple-schema
+theory) and displays entities as tables with their attributes listed inside.
+*/
+export default function SchemaERD(
+    props: ModelAnalysisProps<SchemaERDConfig> & {
+        title?: string;
+    },
+) {
+    const graph = () => {
+        const model = props.liveModel.elaboratedModel();
+        if (model) {
+            return schemaToERD(model);
+        }
+    };
+
+    const [svgRef, setSvgRef] = createSignal<SVGSVGElement>();
+    const [vizResource] = createResource(loadViz);
+
+    const svgString = () => {
+        const viz = vizResource();
+        const g = graph();
+        if (!viz || !g) {
+            return undefined;
+        }
+
+        const direction = props.content.direction ?? Direction.Vertical;
+        return viz.renderString(g, {
+            format: "svg",
+            graphAttributes: {
+                rankdir: direction === Direction.Horizontal ? "LR" : "TB",
+            },
+        });
+    };
+
+    const title = () => props.title ?? "Entity-Relationship Diagram";
+    const header = () => (
+        <>
+            <DownloadSVGButton svg={svgRef()} tooltip={`Export the ${title()} as SVG`} size={16} />
+        </>
+    );
+
+    return (
+        <div class="graph-visualization-container">
+            <Foldable title={title()} header={header()}>
+                <FormGroup compact>
+                    <SelectField
+                        label="Direction"
+                        value={props.content.direction ?? Direction.Vertical}
+                        onChange={(evt) => {
+                            props.changeContent((content) => {
+                                content.direction = evt.currentTarget.value as Direction;
+                            });
+                        }}
+                    >
+                        <option value={Direction.Horizontal}>{"Horizontal"}</option>
+                        <option value={Direction.Vertical}>{"Vertical"}</option>
+                    </SelectField>
+                </FormGroup>
+            </Foldable>
+            <div class="graph-visualization">
+                <Show when={svgString()}>
+                    {(svg) => (
+                        <div
+                            ref={(el) => {
+                                const svgEl = el.querySelector("svg");
+                                if (svgEl) {
+                                    setSvgRef(svgEl);
```

Comment:
Do you need this ref to the SVG element? You're setting the inner HTML directly, and `svgEl` is used only in the download button, but the SVG string is already available. Perhaps skip `DownloadSVGButton` and just directly download Graphviz's own SVG output?